tests/components/weather/test_smhi.py

```diff
@@ -0,0 +1,338 @@
+"""Test the smhi weather entity."""
+import asyncio
+import logging
+from datetime import datetime, timezone
+
+from unittest.mock import Mock, patch
+import smhi as smhi_api
+
+from homeassistant.components.weather import (
+    ATTR_FORECAST_CONDITION, ATTR_FORECAST_TEMP, ATTR_FORECAST_TIME,
+    ATTR_WEATHER_WIND_BEARING, ATTR_WEATHER_WIND_SPEED, smhi)
+from homeassistant.const import STATE_UNKNOWN, TEMP_CELSIUS
+
+from tests.common import mock_coro, mock_coro_func
+
+_LOGGER = logging.getLogger(__name__)
+
+TEST_CONFIG = {
+    "name": "0123456789ABCDEF",
+    "longitude": "62.0022",
+    "latitude": "17.0022"
+}
+
+
+async def test_async_setup_entry() -> None:
+    """Test that we do not discover anything or try to set up a bridge."""
+    hass = Mock()
+    hass.states.async_entity_ids = mock_coro_func([])
+    config_entry = Mock()
+    config_entry.data = TEST_CONFIG
+    config_entries = Mock()
+
+    with \
+        patch.object(smhi, 'SmhiWeather') as weather, \
+        patch.object(smhi, 'async_generate_entity_id',
+                     return_value='some_id'):
+
+        assert await smhi.async_setup_entry(hass, config_entry,
+                                            config_entries) is True
+
+        # Test that config_entries was called correctly
+        called_argument, = config_entries.call_args[0]
+        config_entries.assert_called_with([weather()])
+        assert called_argument[0].entity_id == 'some_id'
+
+        # Test that the constructor parameters of SmhiWeather
+        _, name, latitude, longitude = weather.call_args_list[0][0]
+        assert name == '0123456789ABCDEF'
+        assert longitude == '62.0022'
+        assert latitude == '17.0022'
+
+
+# pylint: disable=W0212
+def test_constructor() -> None:
+    """Test the constructor of SmhiWeather."""
+    hass = Mock()
+    weather = smhi.SmhiWeather(
+        hass, 'name', '17.0022', '62.0022')
+
+    assert weather._latitude == '17.0022'
+    assert weather._longitude == '62.0022'
+    assert weather.name == 'name'
+
+
+def test_properties() -> None:
+    """Test the properties is returned correctly."""
+    hass = Mock()
+    data = Mock()
+    data.temperature = 5
+    data.humidity = 5
+    data.wind_speed = 10
+    data.wind_direction = 180
+    data.horizontal_visibility = 6
+    data.pressure = 1008
+    data.cloudiness = 52
+    data.symbol = 1
+    data.valid_time = datetime(2018, 1, 1, 0, 1, 2)
+
+    testdata = [
+        data,
+        data
+    ]
+
+    weather = smhi.SmhiWeather(hass, 'name', '10', '10')
+    weather._forecasts = testdata
+    assert weather.name == 'name'
```

Comment:
Instead of testing each entity method or property separately in isolation, set up the platform via the home assistant core, while patching the smhi api appropriately, and add the entity, and then get the entity state from the core state machine and assert that the state and state attributes are correct.

This will cut down on the number of required tests and also make the tests more robust.

Reply:
Ok it makes sense to use some kind of "integrationtesting" through the core. That is good feedback. I just need to find out how to do that not having "test config yaml". I only use config_flow.